devito/passes/clusters/blocking.py

```diff
@@ -247,23 +255,18 @@ def callback(self, clusters, prefix):
             if SKEWABLE not in c.properties[d]:
                 return clusters
 
-            if d is c.ispace[-1].dim and not self.skewinner:
-                return clusters
-
             skew_dims = {i.dim for i in c.ispace if SEQUENTIAL in c.properties[i.dim]}
             if len(skew_dims) > 1:
                 return clusters
             skew_dim = skew_dims.pop()
 
-            # The level of a given Dimension in the hierarchy of block Dimensions, used
-            # to skew over the outer level of loops.
-            level = lambda dim: len([i for i in dim._defines if i.is_Incr])
-
             # Since we are here, prefix is skewable and nested under a
             # SEQUENTIAL loop.
             intervals = []
             for i in c.ispace:
-                if i.dim is d and level(d) <= 1:  # Skew only at level 0 or 1
+                if i.dim is d and not d.is_Incr:  # Skew if non-Incr
```

Comment:
how is this different than (as I said on Slack)

```
if i.dim is d and (not d.is_Incr or d._depth == 1):
    intervals.append(Interval(d, skew_dim, skew_dim))
else:
    intervals.append(i)
```

Reply:
I was dumb and used `and` instead of `or` when discussing on slack.. thanks!